gcip/addons/container/config.py

```diff
@@ -0,0 +1,135 @@
+        store itself, mostly you do not want to set the `credsStore`.
+        Use `credsHelpers` instead.
+
+        Args:
+            creds_store (str): Should be the suffix of the program to use
+            (i.e. everything after docker-credential-).
+            For example `osxkeychain`, to use docker-credential-osxkeychain or
+            `ecr-login`, to use docker-crendential-ecr-login
+        """
+        self.config["credsStore"] = creds_store
+
+    def add_cred_helper(self, registry: str, cred_helper: str) -> None:
+        """
+        Adds a Credentials helper `credHelpers` for a registry.
+        See [docker login#credential-helpers](https://docs.docker.com/engine/reference/commandline/login/#credential-helpers)
+
+        Args:
+            registry (str): Name of the container registry to set `creds_helper` for.
+            cred_helper (str): Name of the credential helper to use together with the `registry`.
+        """
+        compose = {
+            registry: cred_helper
+        }
+        if "credsHelpers" not in self.config.keys():
+            self.config["credHelpers"] = compose
+        else:
+            self.config["credHelpers"].update(compose)
+
+    def add_auths(
+        self,
+        registry: str,
+        username_env_var: str = "REGISTRY_USERNAME",
+        password_env_var: str = "REGISTRY_PASSWORD",
+    ) -> None:
+        """
+        Adds basic authentication `auths` setting to the configuration.
+
+        This method acts a little special, because of some security aspects.
+        The method, takse three arguments, `registry`, `username_env_var` and `password_env_var`.
+        Arguments ending wit *_env_var, are ment to be available as a `gcip.Job` variable.
+
+        Args:
+            registry (str): Name of the container registry to set `creds_helper` for.
+            username_env_var (str): Name of the environment variable which as the registry username stored. Defaults to REGISTRY_USERNAME.
+            password_env_var (str): Name of the environment variable which as the registry password stored. Defaults to REGISTRY_PASSWORD.
+        """
+        compose = {
+            registry: {
+                "username": "$" + username_env_var,
+                "password": "$" + password_env_var
+            }
+        }
+
+        if "auths" not in self.config.keys():
+            self.config["auths"] = compose
+        else:
+            self.config["auths"].update(compose)
+
+    def add_raw(self, raw_input: Dict[Any, Any]) -> None:
+        """
+        Adds arbitrary settings to configuration.
+
+        Be aware and warned! You can overwrite any predefined settings with this method.
+        This method is intendet to be used, if non suitable method is available and you
+        have to set a configuration setting.
+
+        Args:
+            raw_input (Dict[Any, Any]): Dictionary of non-available settings to be set.
+        """
+        self.config.update(raw_input)
```

Comment:
Should we add a simple helper script for deep merging dicts? Somethins like
https://gist.github.com/angstwad/bf22d1822c38a92ec0a9
? Would not only be useful here.

Reply:
Mhh yes, would be great. It seems, that the original code has some quirks. e.g. not handling Lists while merging [Comment](https://gist.github.com/angstwad/bf22d1822c38a92ec0a9#gistcomment-2652621). Second could the GPL be a problem? I do not know. The gist is for python2 not python3, there are other contributions in the gist, but as far as I understand neither of them is addressing our problem.
Maybe we should write our own?

Reply:
Opened a tracking issue #43